service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java

```diff
@@ -113,10 +113,19 @@
                                     structuralVariant.setSite2EntrezGeneId(fusion.getEntrezGeneId());
                                     structuralVariant.setSite2HugoSymbol(fusion.getGene().getHugoGeneSymbol());
                                 } else {
+                                    Gene site1Gene = getGene(site1GeneSymbol);
                                     Gene site2Gene = getGene(site2GeneSymbol);
-                                    if (site2Gene != null) {
-                                        structuralVariant.setSite2EntrezGeneId(site2Gene.getEntrezGeneId());
-                                        structuralVariant.setSite2HugoSymbol(site2Gene.getHugoGeneSymbol());
+                                    // we need to pick the right gene since the fusion gene has been assigned to structural variant site1
+                                    Gene pickedGene = null;
+                                    if (site1Gene != null && !site1Gene.getEntrezGeneId().equals(fusion.getEntrezGeneId())) {
+                                        pickedGene = site1Gene;
+                                    }
+                                    if (site2Gene != null && !site2Gene.getEntrezGeneId().equals(fusion.getEntrezGeneId())) {
```

Comment:
shouldn't this be in else block or `pickedGene == null` needs to be included in if caluse

Reply:
@kalletlak updated